util-images/probes/pkg/dnsPropagation/probe.go

```diff
@@ -0,0 +1,242 @@
+		klog.Fatalf("Failed to build kubeClient, error:%v", err)
+	}
+	//TODO deprecated as of Go 1.20. To remove when go version gets bumped
+	rand.Seed(time.Now().UnixNano())
+	runProbe(clientset)
+	for {
+		klog.V(2).Infof("dns propagation probe complete, waiting until the test finishes...")
+		time.Sleep(10 * time.Second)
+	}
+}
+func runProbe(client kubernetes.Interface) {
+	klog.V(1).Infof("DNS propagation probe started")
+	var namespace string
+	for {
+		klog.V(2).Infof("Get namespace from client based on prefix %s", *namespacePrefix)
+		fullNamespace, err := getNamespacesFromClient(client, *namespacePrefix)
+		if err != nil {
+			klog.Warningf("Namespace not found %v", err)
+			time.Sleep(5 * time.Second)
+			continue
+		}
+		klog.V(2).Infof("Found target namespace %v", fullNamespace)
+		namespace = fullNamespace
+		break
+	}
+	var wg sync.WaitGroup
+	ch := make(chan DNSPodPropagationResult)
+	indices := selectSample(*podCount, *sampleCount)
+	durations := make([]float64, 0, *sampleCount)
+	for _, idx := range indices {
+		podName := fmt.Sprintf("%s-%d", *statefulSet, idx)
+		url := fmt.Sprintf("%s.%s.%s.%s.%s.%s", podName, *service, namespace, "svc", *clusterDomain, *suffix)
+		wg.Add(1)
+		go func(client kubernetes.Interface, url, podName string, namespace string, interval time.Duration) {
+			defer wg.Done()
+			result := runSinglePod(client, url, podName, namespace, interval)
+			ch <- DNSPodPropagationResult{
+				podName:  podName,
+				duration: result,
+			}
+		}(client, url, podName, namespace, *interval)
+	}
+	klog.V(2).Infof("Waiting for all sample pods processes to finish")
+	go func() {
+		wg.Wait()
+		close(ch)
+	}()
+
+	for propagationResult := range ch {
+		labels := prometheus.Labels{
+			"namespace": namespace,
+			"service":   *service,
+			"podName":   propagationResult.podName,
+		}
+		DNSPropagationSeconds.With(labels).Set(propagationResult.duration.Seconds())
+		DNSPropagationCount.With(labels).Inc()
+		durations = append(durations, propagationResult.duration.Seconds())
+	}
+	klog.V(2).Infof("Finished calculating DNS propagation for all sample pods")
+
+	if len(durations) == 0 {
+		klog.Warningf("DNS propagation probe has zero obseravtions")
+		return
+	}
+	sum := 0.0
+	for _, duration := range durations {
+		sum += duration
+	}
+	klog.V(1).Infof("DNS propagation probe finished, total of %v observations, average duration, %v s", len(durations), sum/float64(len(durations)))
+}
+
+func selectSample(podCount int, sampleCount int) []int {
+	indices := make([]int, podCount)
+	for idx := range indices {
+		indices[idx] = idx
+	}
+	rand.Shuffle(len(indices), func(i, j int) { indices[i], indices[j] = indices[j], indices[i] })
+	indices = indices[:sampleCount]
+	return indices
+}
+func runSinglePod(client kubernetes.Interface, url string, podName string, namespace string, interval time.Duration) time.Duration {
+	klog.V(4).Infof("Starting dns propagation calculation for pod %s ...", url)
+	tick := time.NewTicker(interval)
```

Comment:
you need to stop the ticker or it will leak resourcecs

https://pkg.go.dev/time#NewTicker
```suggestion
	tick := time.NewTicker(interval)
	defer tick.Stop()
```

Reply:
thanks, fixed